nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -0,0 +1,36 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! [`DataStore`] methods on Database Metadata.
+
+use super::DataStore;
+use crate::db;
+use crate::db::error::public_error_from_diesel_pool;
+use crate::db::error::ErrorHandler;
+use async_bb8_diesel::AsyncRunQueryDsl;
+use diesel::prelude::*;
+use omicron_common::api::external::Error;
+use omicron_common::api::external::SemverVersion;
+use std::str::FromStr;
+
+impl DataStore {
+    pub async fn database_schema_version(
+        &self,
```

Comment:
I'm intentionally not using auth here, since that might rely on DB structures that aren't populated / have invalid versions.